apps/web/pages/api/trpc/[trpc].ts

```diff
@@ -30,34 +30,64 @@ export default trpcNext.createNextApiHandler({
    * @link https://trpc.io/docs/caching#api-response-caching
    */
   responseMeta({ ctx, paths, type, errors }) {
-    // assuming we have all our public routes in `viewer.public`
-    const allPublic = paths && paths.every((path) => path.startsWith("viewer.public."));
-    // checking that no procedures errored
-    const allOk = errors.length === 0;
-    // checking we're doing a query request
-    const isQuery = type === "query";
-
-    // i18n response depends on request header
-    const nonCacheableQueries = ["viewer.public.i18n"];
-    const isThereANonCacheableQuery = paths?.some((path) => nonCacheableQueries.includes(path));
-    const isThereACacheableQuery = paths?.some((path) => !nonCacheableQueries.includes(path));
-    if (isThereANonCacheableQuery && isThereACacheableQuery) {
-      console.warn(
-        "Cacheable and Non-cacheable queries are mixed in the same request. Not going to cache the request"
-      );
-    }
+    // Some helpers relevant to this function only
+    const ONE_DAY_IN_SECONDS = 60 * 60 * 24;
+    const TWO_HOURS_IN_SECONDS = 60 * 60 * 2;
+
+    // Our response to indicate no caching
+    const noCacheResponse = {};
 
+    // TODO: Confirm this is doing what the original code intended
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore ctx.req is available for SSR but not SSG
-    if (!!ctx?.req && allPublic && allOk && isQuery && !isThereANonCacheableQuery) {
-      // cache request for 1 day + revalidate once every 5 seconds
-      const ONE_DAY_IN_SECONDS = 60 * 60 * 24;
-      return {
-        headers: {
-          "cache-control": `s-maxage=5, stale-while-revalidate=${ONE_DAY_IN_SECONDS}`,
-        },
-      };
+    const isSSR = !!ctx?.req;
+    if (isSSR) {
+      return noCacheResponse;
+    }
+
+    // No caching if we have a non-public path
+    // Assuming we have all our public routes in `viewer.public`
+    if (!paths || !paths.every((path) => path.startsWith("viewer.public."))) {
+      return noCacheResponse;
+    }
+
+    // No caching if we have any procedures errored
+    if (errors.length !== 0) {
+      return noCacheResponse;
+    }
+
+    // Never cache queries
+    if (type !== "query") {
+      return noCacheResponse;
     }
-    return {};
+
+    // TODO: Discuss whether this value achieves what the original comment below describes
```

Comment:
I obviously want to preserve the original caching logic, which was as follows:
```javascript
// cache request for 1 day + revalidate once every 5 seconds
let cacheValue = `s-maxage=5, stale-while-revalidate=${ONE_DAY_IN_SECONDS}`;
```
However I don't think this cache is doing what the comment describes. According the [the docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#response_directives), `s-max-age` is typically meant for CDN's or shared resources, which I don't think applies in this case? I might be wrong, but I feel the correct directive would be `max-age`.

Additionally, I think `max-age` should have the larger value, as `stale-while-revalidate` indicates how long the response is valid during the revalidation process.

Again, I could be completely misunderstanding here - but wanted to mention this.

- [ ] @zomars or @PeerRich to confirm that they want to preserve the original caching value